packages/common/src/resourceHelpers.ts

```diff
@@ -226,95 +231,110 @@ export function copyFilesFromStorageItem(
-                  authentication: storageAuthentication
-                },
-                {
-                  itemId: destinationItemId,
-                  authentication: destinationAuthentication
-                }
-              ).then(result => resolveInfo(result), rejectInfo);
-            }, msLag);
-          });
-
-        case EFileType.Metadata:
-          return new Promise<IUpdateItemResponse>(
-            (resolveMetadata, rejectMetadata) => {
-              setTimeout(() => {
-                copyMetadata(
-                  {
-                    url: filePath.url,
-                    authentication: storageAuthentication
-                  },
-                  {
-                    itemId: destinationItemId,
-                    authentication: destinationAuthentication
-                  }
-                ).then(resolveMetadata, rejectMetadata);
-              }, msLag);
+          return copyFormInfoFile(
+            {
+              url: filePath.url,
+              filename: filePath.filename,
+              authentication: storageAuthentication
+            },
+            {
+              itemId: destinationItemId,
+              authentication: destinationAuthentication
             }
           );
 
-        case EFileType.Resource:
-          return new Promise<IUpdateItemResponse>(
-            (resolveResource, rejectResource) => {
-              setTimeout(() => {
-                copyResource(
-                  {
-                    url: filePath.url,
-                    authentication: storageAuthentication
-                  },
-                  {
-                    itemId: destinationItemId,
-                    folder: filePath.folder,
-                    filename: filePath.filename,
-                    authentication: destinationAuthentication
-                  }
-                ).then(resolveResource, rejectResource);
-              }, msLag);
+        case EFileType.Metadata:
+          return copyMetadata(
+            {
+              url: filePath.url,
+              authentication: storageAuthentication
+            },
+            {
+              itemId: destinationItemId,
+              authentication: destinationAuthentication
             }
           );
       }
     });
 
-    // Wait until all files have been copied
-    Promise.all(awaitAllItems).then(() => resolve(true), reject);
+    // Bundle the resources into a single update because AGO tends to have problems with
+    // many updates in a row to the same item: it claims success despite randomly failing
+    const resourceFilePaths = filePaths.filter(
+      filePath => filePath.type === EFileType.Resource
+    );
+    let zip: any;
+    if (resourceFilePaths.length > 0) {
+      zip = new JSZip();
+      awaitAllItems = awaitAllItems.concat(
+        // Note that AGO imposes a limit of 50 files, which is not checked in this code
+        // https://developers.arcgis.com/rest/users-groups-and-items/add-resources.htm
```

Comment:
if/when you tackle the 50 file limit, hub.js has `chunkArray` (https://esri.github.io/hub.js/api/common/chunkArray/) which seems like something that could be useful here...